cwms-data-api/src/test/java/cwms/cda/data/dao/StreamReachDaoTestIT.java

```diff
@@ -0,0 +1,282 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package cwms.cda.data.dao;
+
+import cwms.cda.api.DataApiTestIT;
+import cwms.cda.api.errors.NotFoundException;
+import static cwms.cda.data.dao.DaoTest.getDslContext;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.stream.Bank;
+import cwms.cda.data.dto.stream.Stream;
+import cwms.cda.data.dto.stream.StreamLocation;
+import cwms.cda.data.dto.stream.StreamLocationNode;
+import cwms.cda.data.dto.stream.StreamNode;
+import cwms.cda.data.dto.stream.StreamReach;
+import cwms.cda.helpers.DTOMatch;
+import fixtures.CwmsDataApiSetupCallback;
+import fixtures.TestAccounts;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import mil.army.usace.hec.test.database.CwmsDatabaseContainer;
+import org.jooq.DSLContext;
+import org.junit.jupiter.api.AfterAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("integration")
+final class StreamReachDaoTestIT extends DataApiTestIT {
+
+    private static final String OFFICE_ID = TestAccounts.KeyUser.SWT_NORMAL.getOperatingOffice();
+    private static final List<String> REACH_IDS = new ArrayList<>();
+    private static final List<Stream> STREAMS_CREATED = new ArrayList<>();
+    private static final List<StreamLocation> STREAM_locationsCreated = new ArrayList<>();
+
+    @BeforeAll
+    public static void setup() {
+        try {
+            for (int i = 0; i < 2; i++) {
+                String testLoc = "TEST_REACH" + i;
+                createLocation(testLoc, true, OFFICE_ID, "STREAM_REACH");
+                REACH_IDS.add(testLoc);
+                createLocation(testLoc + "_UP", true, OFFICE_ID, "STREAM_LOCATION");
+                createLocation(testLoc + "_DOWN", true, OFFICE_ID, "STREAM_LOCATION");
+                String streamId = testLoc + "_STREAM";
+                createAndStoreTestStream(streamId);
+                createAndStoreTestStreamLocation(testLoc + "_UP", streamId, 25.0);
+                createAndStoreTestStreamLocation(testLoc + "_DOWN", streamId, 20.0);
+            }
+        } catch (SQLException e) {
```

Comment:
no need to catch, setup() can throw Exception

Reply:
removed